tools/pipelines/test-perf-benchmarks-using-built-artifacts.yml

```diff
@@ -0,0 +1,376 @@
+          inputs:
+            workingFile: ${{ variables.testWorkspace }}/.npmrc
+
+        # Auth to Office feed
+        - task: npmAuthenticate@0
+          displayName: 'npm authenticate (office feed)'
+          inputs:
+            workingFile: ${{ variables.testWorkspace }}/.npmrc
+            customEndpoint: 'Office Packages - Readonly'
+
+        - task: Bash@3
+          displayName: Print Parameters
+          inputs:
+            targetType: 'inline'
+            script: |
+              echo "
+              Pipeline Parameters:
+                poolBuild=${{ parameters.poolBuild }}
+                Build=${{ parameters.taskBuild }}
+
+              Task Variables:
+                releaseBuild=$(releaseBuildVar)
+                toolAbsolutePath=$(toolAbsolutePath)
+                artifactPipeline=$(artifactPipeline)
+                feed=$(feed)
+                BuildDir=${{ variables.buildDirectory }}
+                TestWorkspace=${{ variables.testWorkspace }}
+
+              Build Params
+                SourceBranch=$(Build.SourceBranch)
+              "
+
+        - task: Bash@3
+          displayName: 'Install aria-logger'
+          inputs:
+            targetType: 'inline'
+            workingDirectory: $(toolAbsolutePath)
+            script: |
+              cp ${{ variables.testWorkspace }}/.npmrc . ;
+              npm install @ff-internal/aria-logger;
+              npm i;
+              npm run build:compile;
+
+        # Download Test Files
+        # Download Artifact - Test Files
+        - task: DownloadPipelineArtifact@2
+          displayName: Download test files
+          inputs:
+            source: specific
+            project: internal
+            pipeline: ${{ variables.artifactPipeline }}
+            preferTriggeringPipeline: true
+            allowPartiallySucceededBuilds: true
+            runVersion: latestFromBranch
+            runBranch: $(Build.SourceBranch)
+            artifact: test-files
+            path: $(Pipeline.Workspace)/test-files/
+
+        - ${{ each execTestPackage in parameters.executionTestPackages }}:
+
+          # Download package artifacts
+          - task: DownloadPipelineArtifact@2
+            displayName: Download test package
+            inputs:
+              source: specific
+              project: internal
+              pipeline: ${{ variables.artifactPipeline }}
+              preferTriggeringPipeline: true
+              allowPartiallySucceededBuilds: true
+              runVersion: latestFromBranch
+              runBranch: $(Build.SourceBranch)
+              artifact: pack
+              patterns: "**/${{ replace(replace(execTestPackage, '@', '' ), '/', '-') }}-*.tgz"
+              path: $(Pipeline.Workspace)/client/pack
+
+          # Install test package
+          - task: Bash@3
+            displayName: Install Test Package
+            inputs:
+              targetType: 'inline'
+              workingDirectory: ${{ variables.testWorkspace }}
```

Comment:
It might be a good idea to isolate the test directory for each package?  or clean up after each package?
Just in case installing those packages has side effects.

Reply:
Letmelook into that!

Reply:
Okay I've added a clean up step [here](https://github.com/microsoft/FluidFramework/pull/12778/files#diff-48aabc0ffe520e67dff3410a6412af4a5a33f17fcd4485d9f09b515ca043d800R269-R276) and [here](https://github.com/microsoft/FluidFramework/pull/12778/files#diff-48aabc0ffe520e67dff3410a6412af4a5a33f17fcd4485d9f09b515ca043d800R371). So, this is good to go!